index.js

```diff
@@ -0,0 +1,50 @@
+const express = require("express");
+const app = express();
+const axios = require("axios");
+const pool = require("./database");
+require("dotenv").config();
+
+app.use(express.json());
+
+let port = process.env.PORT || 3000;
+
+app.listen(port, () => {
+  console.log(`running on ${port}`);
+});
+
+app.post("/github_push_webhook/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+    var reqData = req.body;
+
+    const commit_message = reqData.head_commit.message || reqData.repository.id;
+    // using this because when github sends the first POST request, there is no commit message which leads to error 400
+
+    if (
+      reqData.repository.owner.login === null ||
+      commit_message === null ||
+      reqData.repository.name === null
+    ) {
+      return res.status(400).json({ error: "bad format" });
+    }
+
+    var message = `${reqData.repository.owner.login} just pushed a commit with message - '${commit_message}' to <${reqData.repository.name}>`;
+    const data = await pool.query(
+      "select discord_urls from github_discord_url where id = $1",
+      [id]
+    );
+    const discord_url = data.rows[0];
```

Comment:
check case where thee were no rows returned 
& 
check case where there were more than 1 row is returned. 
Both these cases indicate that something went wrong

Reply:
how will there be no rows returned, the field is compulsory and can't be left null

Reply:
I don't get the cases yet

Reply:
let's talk regarding this issue